### DISCUSSION TITLE
Custom method when action recieved

### DISCUSSION THREAD

**Discussion by Stips5** [timmbogner/Farm-Data-Relay-System#180] [Ideas]
Hi

I am trying to optimize one part of system in a way that I have node attached to gateway, but node should be triggered to make reading when triggered by gateway sending HIGH to CH_PD pin which would wake it up, and its execution will start and it will send message to gateway through defined protocol. But problem or question is: could it be possible to have some kind of callback handler, so when some of actions is ACT, do something else than send it to serial or lora or something, I would live to call custom function in this case.

**Comment by timmbogner** [comment]
Hey @Stips5 ,

I think this is possible, but I need to make sure we're talking about the same scenario. 

For example: You want to set a pin high every time ESPNOWG_ACT is triggered, correct?

If so, you can just add the desired `digitalWrite()` code right into the "ESPNOWG_ACT" line. There's probably a way to have a function that you can add to the ACT lines, but I think you would need to declare it before the header files. Apologies, as I'm a bit rusty on the compile order details and am away from my workshop.

Let me know if this makes sense and/or is helpful!

**Reply by Stips5** [reply]
Should hsve checked digitalWrite() in line to ESP_ACK for example, but i think some custom method would also be nice to have. Or maybe there coul be set flag which could be checked throught some getter method, if its some trouble to mess around with compiling.

**Reply by timmbogner** [reply]
I drew up an example of how this should work. The config could look like this : ``` #define ESPNOWG_ACT sendSerial(); enablePin(13); ```

And here is the main sketch:
``` cpp
//  FARM DATA RELAY SYSTEM
//
//  GATEWAY 2.000
//

void enablePin(uint8_t pin) {
  digitalWrite(pin, HIGH);
  delay(10);
  digitalWrite(pin, LOW);
}


#include "fdrs_gateway_config.h"
#include <fdrs_gateway.h>

void setup() {
  beginFDRS();
}

void loop() {
  loopFDRS();
}
```
